Improve listeners performances

MineTheCube opened the pull request:
This should slightly improve performances, especially with a lot of shops.
Now shops are updated only on block changes and not every move event.
It also uses distanceSquared instead of distance to avoid sqrt operations.

EpicEricEE commented on src/main/java/de/epiceric/shopchest/listeners/HologramUpdateListener.java:
```diff
@@ -20,8 +20,17 @@ public HologramUpdateListener(ShopChest plugin) {
 
     @EventHandler
```
And as you added the following to the PlayerMoveEvent in the ShopItemListener, you can also add it here.
```java
@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
```

MineTheCube commented:
Ok so I've updated annotation on `ShopItemListener#onPlayerMove(PlayerMoveEvent)` and renamed both methods to `updateShopItemVisibility`.

But argument `playerLocation` is needed for the `PlayerTeleportEvent`, as location used is `e.getTo()` and not current player location.

EpicEricEE commented:
Ok, then everything is fine. Thank you very much :+1:
